source/common/router/upstream_request.cc

```diff
@@ -406,6 +406,8 @@ void UpstreamRequest::onPoolFailure(ConnectionPool::PoolFailureReason reason,
     reset_reason = Http::StreamResetReason::ConnectionFailure;
   }
 
+  stream_info_.upstreamInfo()->setUpstreamTransportFailureReason(transport_failure_reason);
+
```

Comment:
May be move this to the `onResetStream` where will set some other flgas to stream info?

Reply:
I dont understand. The idea here is to propagate the transport failure reason in the context on `onPoolFailure`. Since the reason is available in the callback `onPoolFailure`, we need to capture that here.